core/src/main/kotlin/org/evomaster/core/search/service/SearchAlgorithm.kt

```diff
@@ -30,16 +30,45 @@ abstract class SearchAlgorithm<T> where T : Individual {
     @Inject
     protected lateinit var config: EMConfig
 
-
     @Inject(optional = true)
     private lateinit var mutator: Mutator<T>
 
+    private var lastSnapshot = 0
 
-    protected fun getMutatator() : Mutator<T> {
+    protected fun getMutator(): Mutator<T> {
         return mutator
     }
 
-    abstract fun search() : Solution<T>
+    abstract fun searchOnce()
+
+    abstract fun setupBeforeSearch()
+
+    fun search(writeTests: (s: Solution<T>, t: String) -> Unit): Solution<T> {
```

Comment:
can u add JavaDoc explaining what `writeTests` is?
in particular, it is unclear what `t` in the lambda is supposed to do

Reply:
furthermore, in a lot of place you call `.search { _: Solution<*>, _: String ->  }`, which I assume is simply a lambda that tells that the snapshot printing does nothing

in that case, it would make sense to make `writeTests` nullable, and already initialized with default parameter `= null`. 
This would allow to simply call `.search()` when no snapshot is required